src/targets/gpu/pack_int8_args.cpp

```diff
@@ -69,6 +93,76 @@ shape pack_int8_args::pack_int8_shape(const shape& s) const
     return {s.type(), lens, strides};
 }
 
+instruction_ref pack_int8_args::pad_ins(module& p, instruction_ref ins, int offset) const
+{
+    auto s                         = ins->get_shape();
+    auto lens                      = s.lens();
+    auto k                         = lens[lens.size() + offset];
+    auto pad_k                     = (k + 3) / 4 * 4;
+    auto pad_lens                  = lens;
+    pad_lens[lens.size() + offset] = pad_k;
+    std::vector<int64_t> pad_dims(lens.size() * 2, 0);
+    auto ret_ins = ins;
+    if(pad_k != k)
+    {
+        pad_dims[lens.size() + offset] = pad_k - k;
+        shape ps{s.type(), pad_lens};
+        auto ins_out = p.insert_instruction(ins, hip_allocate{ps});
+        op::pad pad{pad_dims};
```

Comment:
Use `make_op` to construct the pad operator.

Reply:
changed